pkg/gcp/secret.go

```diff
@@ -0,0 +1,56 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/secretmanager/v1"
+)
+
+const secretPrefix = "ghatoken"
+
+// SecretClient is GCP compute client with go client
+type SecretClient struct {
+	service   *secretmanager.Service
+	projectID string
+}
+
+// NewSecretClient returns SecretClient
+func NewSecretClient(
+	ctx context.Context,
+	projectID string,
+) (*SecretClient, error) {
+	s, err := secretmanager.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SecretClient{
+		service:   s,
+		projectID: projectID,
+	}, nil
+}
+
+func (c *SecretClient) GetSecret(repoName string) (*secretmanager.Secret, error) {
+	return c.service.Projects.Secrets.Get(fmt.Sprintf("projects/%s/secrets/%s_%s", c.projectID, secretPrefix, repoName)).Do()
+}
+
+func (c *SecretClient) CreateSecretFromData(expireTime string, repoName string, data string) error {
+	parent := "projects/" + c.projectID
+	secretName := fmt.Sprintf("%s/secret/%s_%s", parent, secretPrefix, repoName)
```

Comment:
It is good to add the prefix by the caller of this function.

The struct name `SecretClient` and the function name `CreateSecretFromData` remind me this function can use something generic purpose.
However, the prefix `ghatoken_` would limit its usage.